pandas/tools/merge.py

```diff
@@ -202,10 +208,41 @@ def get_result(self):
         typ = self.left._constructor
         result = typ(result_data).__finalize__(self, method='merge')
 
+        if self.indicator:
+            # Has side-effects
+            self._indicator_post_merge(result, self.left, self.right)
+
         self._maybe_add_join_keys(result, left_indexer, right_indexer)
 
         return result
 
+    def _indicator_pre_merge(self, left, right):
+        
+        columns = left.columns.values.tolist() + right.columns.values.tolist()  
+
+        for i in ['_left_indicator', '_right_indicator', '_merge']:
+            if i in columns:
+                raise ValueError("Cannot use `indicator=True` option when data contains a column named {}".format(i))
+
+        left['_left_indicator'] = 1  
+        left['_left_indicator'] = left['_left_indicator'].astype('int8')  
+        
+        right['_right_indicator'] = 2     
+        right['_right_indicator'] = right['_right_indicator'].astype('int8')  
+
+
+    def _indicator_post_merge(self, result, left, right):
+        result['_left_indicator'].fillna(0, inplace=True)        
```

Comment:
don't use ``inplace``. It makes the code much harder to read and is not used anywhere within the pandas codebase (well it shouldn't be)